src/components/widgets/accounts-cash-projection/accounts-cash-projection.directive.coffee

```diff
@@ -32,29 +34,81 @@ module.controller('WidgetAccountsCashProjectionCtrl', ($scope, $q, $filter) ->
         startOnTick: false
         minPadding: 0
         tickInterval: 1
-        labels:
-          formatter: ->
-            $filter('mnoDate')(w.content.chart.labels[this.value], getPeriod())
-        plotLines: [{
-          color: 'rgba(0, 85, 255, 0.2)'
-          value: getTodayMarker()
-          width: 1
-          label:
-            text: null
-            verticalAlign: 'top'
-            textAlign: 'center'
-            rotation: 0
-            y: -5
-        }]
       yAxis:
         title: null
         startOnTick: true
         minPadding: 0
-        labels:
-          formatter: ->
-            $filter('mnoCurrency')(this.value, w.metadata.currency, false, 0)
+    data: ->
       series: w.content.chart.series
-    }
+    formatters: ->
+      xAxis:
+        labels:
+            formatter: ->
+              $filter('mnoDate')(w.content.chart.labels[this.value], getPeriod())
+      yAxis:
+        labels:
+            formatter: ->
+              $filter('mnoCurrency')(this.value, w.metadata.currency, false, 0)
+    todayMarker: ->
+      xAxis:
+        plotLines: [{
+            color: 'rgba(0, 85, 255, 0.2)'
+            value: getTodayMarker()
+            width: 1
+            label:
+              text: null
+              verticalAlign: 'top'
+              textAlign: 'center'
+              rotation: 0
+              y: -5
+          }]
+    thresholdMarker: ->
+      yAxis:
+        plotLines: [{
+          color: 'rgba(255, 0, 0, 0.5)'
+          value: getThresholdTarget()
+          width: 2
+          zIndex: 5
+        }]
+  }
+
+  # Widget specific methods
+  # --------------------------------------
+  w.initContext = ->
+    $scope.isDataFound = w.content?
+    initChart()
+
+  # Can be removed once click even is registered in the attach-kpi cmp
+  $scope.onAttachKpiInit = ({api})->
+    $scope.settingAttachKpiApi = api
+
+  $scope.onAttachedKpi = ({kpi})->
+    $scope.chart.update(chartFactory.thresholdMarker())
+
+
+  # Private
+
+  initChart = ->
```

Comment:
There is already a `widget.format()` that is automatically called after `.initContext`
=> check the PR on Simulation mode / I've refactored that bit